neqo-http3/src/recv_message.rs

```diff
@@ -161,12 +172,26 @@ impl RecvMessage {
             return Err(Error::HttpGeneralProtocolStream);
         }
 
-        self.conn_events
-            .header_ready(self.stream_id, headers, interim, fin);
+        let is_web_transport = self.message_type == MessageType::Request
+            && headers
+                .iter()
+                .any(|h| h.name() == ":method" && h.value() == "CONNECT")
```

Comment:
An idea that might be worth doing in future, but maybe it's worth it anyway...

When we look for pseudo-header fields, it would be much faster if we stopped looking once we got a non-pseudo-header field...

```rust
headers.iter().take_until(|h| h[0] != ':').any(|h| ....)
```

Note:
1. This assumes that we validate headers and ensure that !h.is_empty() somewhere.
1. That also speaks to having a struct for headers rather than a dumb `Vec<Header>`.